tests/otel_tracing_e2e/test_e2e.py

```diff
@@ -0,0 +1,83 @@
+import json
+import os
+
+from opentelemetry import trace, propagate
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.resources import DEPLOYMENT_ENVIRONMENT, Resource, SERVICE_NAME
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import BatchSpanProcessor
+from opentelemetry.semconv.trace import SpanAttributes
+from opentelemetry.propagators.b3 import B3MultiFormat, B3SingleFormat
+from _validator import validate_trace
+from utils import context, weblog, interfaces, scenarios, missing_feature
+
+
+class E2ETestBase:
+    def setup_main(self):
+        self.setup_opentelemetry()
+        self.use_128_bits_trace_id = False
+        with self.tracer.start_as_current_span(name="runner.get", kind=trace.SpanKind.CLIENT) as span:
+            span.set_attribute(SpanAttributes.HTTP_METHOD, "GET")
+            span.set_attribute(SpanAttributes.HTTP_URL, "http://weblog:7777")
+            headers = {}
+            propagate.get_global_textmap().inject(headers)
+            self.r = weblog.get(path="/", headers=headers)
+            span.set_attribute(SpanAttributes.HTTP_STATUS_CODE, self.r.status_code)
+            self.trace_id = span.get_span_context().trace_id
+            self.root_span_id = span.get_span_context().span_id
+
+    def setup_opentelemetry(self):
```

Comment:
Probably you will want to move this logic into a shared helper outside the test class if you will be adding more tests later on since I imagine this will be the same setup needed by every OTel test right (e.g. figuring out dd site URLs, setting up the exporter, etc)?

Reply:
Per discussion with Charles, it seems an anti-pattern to instrument the test container, so I removed all the OTel client-side instrumentation and dependencies.